test/model/test/model_transaction/main.cpp

```diff
@@ -352,6 +365,308 @@ test_transaction_basic_wt(void)
     verify_using_debug_log(opts, test_home.c_str(), true);
 }
 
+/*
+ * test_transaction_column_wt --
+ *     The basic test of the transaction model with a column store, also in WiredTiger.
+ */
+static void
+test_transaction_column_wt(void)
+{
+    model::kv_database database;
+
+    model::kv_table_config table_config;
+    table_config.type = model::kv_table_type::column;
+    model::kv_table_ptr table = database.create_table("table", table_config);
+
+    /* Transactions. */
+    model::kv_transaction_ptr txn1, txn2;
+
+    /* Create the test's home directory and database. */
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    WT_SESSION *session1;
+    WT_SESSION *session2;
+    const char *uri = "table:table";
+
+    std::string test_home = std::string(home) + DIR_DELIM_STR + "column";
+    testutil_recreate_dir(test_home.c_str());
+    testutil_wiredtiger_open(opts, test_home.c_str(), ENV_CONFIG, nullptr, &conn, false, false);
+    testutil_check(conn->open_session(conn, nullptr, nullptr, &session));
+    testutil_check(conn->open_session(conn, nullptr, nullptr, &session1));
+    testutil_check(conn->open_session(conn, nullptr, nullptr, &session2));
+    testutil_check(
+      session->create(session, uri, "key_format=r,value_format=S,log=(enabled=false)"));
+
+    /* A basic test with two transactions. */
+    wt_model_txn_begin_both(txn1, session1);
+    wt_model_txn_begin_both(txn2, session2);
+    wt_model_txn_insert_both(table, uri, txn1, session1, recno1, value1);
+    wt_model_txn_insert_both(table, uri, txn2, session2, recno2, value2);
+    wt_model_txn_assert(table, uri, txn1, session1, recno1);
```

Comment:
Beside the below I have no comments, though I did read all changes.

I think it would be easier to read the test to see if the correct thing is tested if the expected value is given. In wt_model_txn_assert() _ret is used for the return value and _out for the result. Thus it could be done this way:
`
/*
 * wt_model_assert_with_expected --
 *     Check that the key has the same value in the model as in the database and as expected.
 */
#define wt_model_assert(table, uri, key, ret_expected, value_expected, ...)                              \
    {                                                                      \
        model::data_value __out_model, __out_wt, __out_expected;                           \
        int __ret_model, __ret_wt, __ret_expected;                                         \
        __ret_model = table->get_ext(key, __out_model, ##__VA_ARGS__);     \
        __ret_wt = wt_get_ext(session, uri, key, __out_wt, ##__VA_ARGS__); \
        testutil_assert(__ret_model == __ret_wt);                          \
        testutil_assert(__out_model == __out_wt);                          \
        wt_model_assert_equal(__ret_model, __ret_wt);                      \
        wt_model_assert_equal(__out_model, __out_wt);                      \
        __ret_expected = ret_expected; \
        __out_expected = value_expected; \
        wt_model_assert_equal(__ret_model == __ret_expected); \
        wt_model_assert_equal(__out_model == __out_expected); \
    }
`
This would be slower if the comparison __out_model == __out_expected is slow like comparing 8MB strings. Thus wt_model_txn_assert() could still be used for any test cases like that.